elyra/pipeline/component_parameter.py

```diff
@@ -169,129 +262,132 @@ def add_to_execution_object(self, runtime_processor: RuntimePipelineProcessor, e
         """
         pass
 
-    def get_all_validation_errors(self) -> List[str]:
-        """Perform custom validation on an instance."""
-        return []
-
-    def is_empty_instance(self) -> bool:
-        """Returns a boolean indicating whether this instance is considered a no-op."""
-        return False
-
 
 class DisableNodeCaching(ElyraProperty):
-    """Disable caching to force node re-execution in the target runtime environment."""
+    """An ElyraProperty representing node cache preference"""
+
+    applies_to_generic = False
+    applies_to_custom = True
```

Comment:
Where are these used?  I only see them being set.

Reply:
These are used in `get_classes_for_component_type` in the same file (in line 188 right now). If a property has `applies_to_{component_type} == True`, then it is added to the list of properties to render for that component. I had these attributes just named `generic` and `custom` up until a few commits ago, but thought that that name maybe wasn't descriptive enough.

Reply:
Ah - thanks.  I had searched for the complete variable name and didn't see its templated use.  Thank you.